Chart/Chart.swift

```diff
@@ -451,6 +463,38 @@ class Chart: UIControl {
         return lineLayer
     }
     
+    private func drawPoints(xValues xValues: Array<Float>, yValues: Array<Float>, seriesIndex: Int, i: Int) -> CAShapeLayer {
+        
+        let circleLayer = CAShapeLayer()
+        let circleRadius: CGFloat = 2.0
+        
+        func circleFrame(x: CGFloat, y: CGFloat) -> CGRect {
```

Comment:
Not a good idea declaring a function inside a method which is looped over a relative big amount of values. I'd put the shape layer outside the class.